### PR TITLE
refactor(artifacts): remove duplicate 'name' parameter in _DownloadedArtifactEntry

### PR DESCRIPTION
**Pull request by moredatarequired**
Description
-----------
In a `_DownloadedArtifactEntry`, self.name is _always_ equal to self.path, so it's a superfluous parameter. No big change, this just removes it; note that this is entirely internal, so there's no user-facing API implications.

### <samp>🤖 Generated by Copilot at e7480be</samp>

> _To refactor some code was the goal_
> _Of this pull request that made it whole_
> _It removed `name` from `_DownloadedArtifactEntry`_
> _And used `path` instead, more sensibly_
> _It also improved `ArtifactManifestEntry`'s doc and style role_

### CODE REVIEW

**Review by szymon-piechowicz-wandb — APPROVED** [review]

File: wandb/apis/public.py [hunk 1]
```diff
@@ -4125,7 +4124,6 @@ def __init__(
             extra=entry.extra,
             local_path=entry.local_path,
         )
-        self.name = name
```
[the remark this answers is not in the document]
Comment (moredatarequired) [reply wandb/apis/public.py:4128]: How do you feel about adding a `@property` for name that returns `entry.path`? The only reason I would want to do that is either to log a warning that this is deprecated and they should use `entry.path` instead, and/or to send back telemetry to see if in fact we have any users using `entry.name` for something.

[the remark this answers is not in the document]
Comment (szymon-piechowicz-wandb) [reply wandb/apis/public.py:4128]: Sure, that sounds good.

[the remark this answers is not in the document]
Comment (vwrj) [reply wandb/apis/public.py:4128]: Nice catch Szymon, adding a `@property` sounds good. LGTM otherwise!

**Review by vwrj — APPROVED** [review]
Adding `@property` per Szymon's comment should do it :)

File: wandb/apis/public.py [hunk 2]
```diff
@@ -4125,9 +4124,14 @@ def __init__(
...
         self._parent_artifact = parent_artifact
 
+    @property
+    def name(self):
+        # TODO(hugh): add telemetry to see if anyone is still using this.
+        wandb.termwarn("ArtifactManifestEntry.name is deprecated, use .path instead")
```
Comment (moredatarequired) [inline wandb/apis/public.py:4132]: Is it okay if I add a term warning and a TODO to actually add telemetry? @szymon-piechowicz-wandb @vwrj

Comment (szymon-piechowicz-wandb) [reply wandb/apis/public.py:4132]: lgtm

Comment (vwrj) [reply wandb/apis/public.py:4132]: sg

**Pull request merged**